### PR TITLE
OverrideScalaMediator: fix bootstrapping a Scala compiler

### LINKED ISSUE #2867: autoScalaLibrary := false is ignored
From https://github.com/xeno-by/eden/commit/e5c1d1a5a8511cc3998099b27c46c5e9edd33e3b#commitcomment-20183004

It would appear that something in sbt 0.13.12 makes sbt "disregards the option `autoScalaLibrary := false`".

Further investigation is required. @liufengyun would you willing to try and provide a minimised reproduction of the issue you saw?

### PR DESCRIPTION
**Pull request by smarter**
Fixes #2867, #2896

### CODE REVIEW

**Comment by smarter** [comment]
To test this with http://github.com/smarter/dotty-example-project or any other dotty project you'll need to wait forhttps://github.com/lampepfl/dotty/pull/1889 to be merged.

File: ivy/src/main/scala/sbt/IvyScala.scala [hunk 1]
```diff
@@ -64,13 +64,22 @@ private object IvyScala {
       // Mediate only for the dependencies in scalaVersion configurations. https://github.com/sbt/sbt/issues/2786
       def configQualifies: Boolean =
         (dd.getModuleConfigurations exists { scalaVersionConfigs })
+      // Do not rewrite the dependencies of Scala dependencies themselves, this prevents bootstrapping
+      // a Scala compiler using another Scala compiler.
```
Comment (eed3si9n) [inline ivy/src/main/scala/sbt/IvyScala.scala:68]: Interesting.

**Review by eed3si9n — APPROVED** [review]
Thanks for the fix!

**Comment by smarter** [comment]
I can't reproduce the `sbt.internals.parser.SplitExpressionsTest` test failure so I assume it's transient, I can reproduce the `scripted dependency-management/snapshot-resolution` failure but I have absolutely no idea how this is related to this PR (or what this test is about), any clue?

**Comment by smarter** [comment]
Magic! Adding a sleep helped.

**Comment by smarter** [comment]
Should be ready to go now, but I can't merge it myself.

**Pull request merged**